src/lasp_core.erl

```diff
@@ -269,6 +269,8 @@ declare(Id, Type, MetadataFun, MetadataNew, Store) ->
             %% Do nothing; make declare idempotent at each replica.
             {ok, {{Id, Type}, Type, Metadata, Value}};
         _ ->
+            %% @todo is this the correct place?
```

Comment:
Yes, I believe this is correct.

However, you also probably should look at the `declare_if_not_found` function as well, which will declare a variable that isn't declared yet on read if the message happened to be delayed for some reason.